state/presence/presence.go

```diff
@@ -1027,3 +1027,34 @@ func beingsC(base *mgo.Collection) *mgo.Collection {
 func pingsC(base *mgo.Collection) *mgo.Collection {
 	return base.Database.C(base.Name + ".pings")
 }
+
+func removeModelFromCollection(coll *mgo.Collection, modelUUID string) error {
+	modelIDMatch := bson.M{"_id": bson.RegEx{"^" + modelUUID + ":", ""}}
+	if changed, err := coll.RemoveAll(modelIDMatch); err != nil {
+		if err == mgo.ErrNotFound {
+			// not a worthy error
+			return nil
+		}
+		return errors.Trace(err)
+	} else {
+		logger.Debugf("removed %d entries from %q for model %q",
+			changed.Removed, coll.FullName, modelUUID)
+	}
+	return nil
+}
+
+// RemovePresenceForModel removes all of the records of entities for a given model
+// across all of the collections.
+func RemovePresenceForModel(base *mgo.Collection, modelTag names.ModelTag) error {
```

Comment:
Other parts of Presence also take names.ModelTag (NewPinger and NewWatcher)
At one point "tags" were certainly just for the API. From reading the code I think the idea is to have "typed strings" so that you don't pass a machine Id vs a unit vs a model. Otherwise they are all just strings.